src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -217,51 +218,44 @@ Literal Literal::numeric_binop_impl(OpSelect op_select, Literal const &other, No
         DatatypeRegistry::NumericOpResult const op_res = op_select(*equalized_entry->numeric_ops)(equalizer->convert_lhs(this->value()),
                                                                                                   equalizer->convert_rhs(other.value()));
 
-        auto const to_string_fptr = [&]() {
-            if (op_res.result_type_iri == equalized_entry->datatype_iri) [[likely]] {
-                return equalized_entry->to_string_fptr;
-            } else [[unlikely]] {
-                return DatatypeRegistry::get_to_string(op_res.result_type_iri);
-            }
-        }();
+        auto const to_string_fptr = op_res.result_type_iri == equalized_iri
```

Comment:
Yes, I agree and wanted to do that the problem is just that lambdas cannot really capture structured bindings by reference or value it needs really ugly workarounds for that to work. The reason is something about structured bindings not being variables. This is how it would look

```
auto const to_string_fptr = [&op_res, equalized_id = std::ref(equalized_id), equalized_entry = equalized_entry]() {
    if (op_res.result_type_id == equalized_id.get()) [[likely]] {
        return equalized_entry->to_string_fptr;
    } else [[unlikely]] {
        return DatatypeRegistry::get_to_string(op_res.result_type_id);
    }
}();
```

Reply:
Ah, right. the structured binding - lambda thing is really annoying. 

I would still stick with the solution with `std::ref` you proposed. I think the likely/unlikely really makes sense here.